ci-operator/step-registry/baremetal/lab/agent/conf/baremetal-lab-agent-conf-chain.yaml

```diff
@@ -6,6 +6,7 @@ chain:
     - ref: baremetal-lab-pre-load-balancer
     - ref: baremetal-lab-pre-dns
     - ref: baremetal-lab-pre-firewall
+    - ref: baremetal-lab-upi-conf-grub2
```

Comment:
Although this is harmless for the other agent workflows, we should not add the grub configuration files for installations not using PXE as boot mode.

As the other agent configurations leverage a dedicated conf chain, we should do the same for pxe: create a new folder, `pxe`, in the lab/agent/conf folder, add this chain (including the step baremetal-lab-upi-conf-grub2) there, and rollback the default one. Also, set the `PXE_MODE` variable to `pxe` in the `chain.environment` stanza, so that we hide this detail to the users.

Reply:
I added it just for testing purposes. Yes, I will create a separate workflow for PXE.